compiler/qsc/src/bin/qsc.rs

```diff
@@ -98,11 +101,10 @@ fn main() -> miette::Result<ExitCode> {
         Ok(ExitCode::SUCCESS)
     } else {
         for error in errors {
-            if let Some(source) = unit.sources.find_by_diagnostic(&error) {
-                eprintln!("{:?}", Report::new(error).with_source_code(source.clone()));
-            } else {
-                eprintln!("{:?}", Report::new(error));
-            }
+            eprintln!(
+                "{:?}",
+                Report::new(WithSource::from_map(&unit.sources, error))
```

Comment:
Essentially I removed all uses of `with_source_code` from the code base. To properly map spans to sources, you must always construct a `WithSource` wrapper, passing the whole `SourceMap` .

Reply:
I think what you have in comments for the PR you should have in comments to the source code. For example, you could put a comment saying that we pass the whole map to make sure we report multi-file spans properly...